### ISSUE TITLE
"patch" returns 127 on Windows

### ISSUE THREAD

**Issue by legal90** [chef/omnibus-toolchain#94]
Recently I found that Windows build machines which were always working fine, started throwing an error wile trying to build a software which haven't been changed for a while too:

```
        [Builder: libffi] I | 2018-06-07T16:06:32+02:00 | Build libffi: 0.4239s
The following shell command exited with status 127:

    $ CFLAGS=-IC:/apica/myproj/embedded/include -m64 -O3 -march=x86-64 CPPFLAGS=-IC:/apica/myproj/embe
dded/include -m64 -O3 -march=x86-64 CXXFLAGS=-IC:/apica/myproj/embedded/include -m64 -O3 -march=x86-64 LDFLAGS=
-LC:/apica/myproj/embedded/lib -m64 -fno-lto LD_RUN_PATH=C:/apica/myproj/embedded/lib OMNIBUS_INSTALL_
DIR=C:/apica/myproj PKG_CONFIG_PATH=C:/apica/myproj/embedded/lib/pkgconfig Path=C:/apica/apicanet-work
er/bin;C:/apica/myproj/embedded/bin;C:/opscode/omnibus-toolchain/embedded/lib/ruby/gems/2.5.0/bin;C:\wix;C:\Pro
gram Files\7-zip;C:\Program Files (x86)\Windows Kits\8.1\bin\x64;C:\opscode\omnibus-toolchain\embedded\bin;C:\opscode\om
nibus-toolchain\embedded\bin\mingw64\bin;C:\opscode\omnibus-toolchain\embedded\bin\usr\bin;C:\opscode\omnibus-toolchain\
embedded\git\cmd;C:\opscode\omnibus-toolchain\embedded\git\mingw64\libexec\git-core;C:\Program Files (x86)\Parallels\Par
allels Tools\Applications;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\
v1.0\;C:\opscode\chef\bin\;c:\opscode\omnibus-toolchain\bin\ bash -c 'patch -p1 -i ../../../../opscode/omnibus-toolchain
/embedded/lib/ruby/gems/2.5.0/bundler/gems/omnibus-software-cb42641d8c0a/config/patches/libffi/libffi-3.2.1-disable-mult
i-os-directory.patch'

Output:

    (nothing)

Error:

    bash.exe: warning: could not find /tmp, please create!
mkdir: cannot change permissions of '/dev/shm': Permission denied
mkdir: cannot change permissions of '/dev/mqueue': Permission denied
```
`libffi` software is taken from here, it wasn't changed almost for 2 years:
https://github.com/chef/omnibus-software/blob/master/config/software/libffi.rb

On the subsequent run it does the same, but the error output become empty. After some investigation I found that `patch` executable stopped working there:
```
PS> bash

vagrant@WIN-GSSPMDMT59Q MINGW64 /c/vagrant/myproj
$ patch --help

vagrant@WIN-GSSPMDMT59Q MINGW64 /c/vagrant/myproj
$ echo $?
127
```
![builder-windows-2012r2_default_1528383012504_93853_and_control_center](https://user-images.githubusercontent.com/4846154/41108933-170e25cc-6a76-11e8-9927-f224117bdf79.jpg)

I've reproduced that on or CI build machines and locally via TestKitcehn too. So, I guess that this issue is global, but I don't understand what actually caused that.
Any help is appreciated!

### Environment details
Host OS: Windows 2012 R2
omnibus-toolchain package version: 1.1.88

The TestKitchen environment was converged by `omnibus::default` recipe of `omnibus` cookbook v5.5.0 with no attributes overridden.

**Comment by legal90** [comment]
I found that this issue doesn't exist in previous releases in both channels:
- `1.1.83` channel `current`
- `1.1.82` channel `stable`
In both cases `patch.exe` works fine in the same environment. 

So it seems that the issue was introduced somewhere in https://github.com/chef/omnibus-toolchain/compare/1.1.83...1.1.88

**Comment by scotthain** [comment]
@legal90 If I had to guess, I'd say that MSYS patch was updated and has a bug in it. Since MSYS uses pacman for their internal package manager (which the toolchain wraps) we only get a "snapshot" of what it looks like at the time of toolchain build. (There's no way to pin to the previous version, for example). We'll be looking into this asap.

**Comment by legal90** [comment]
@scotthain Thanks!
I also want to notice that I tried to build `omnibus-toolchain` msi by myself using a TestKitchen instance with `angry-omnibus-toolchain` v1.1.82 installed (because later versions of it has the same issue with `patch`, they can be used for building `omnibus-toolchain`). 

The interesting fact is that _all_ versions of `omnibus-toolchain` produced by such builder were having this bug incide. 
At least I reproduced it by building msi's from these tags: `1.1.88`, `1.1.83`, `1.1.82`. That's weird...
